Small refactoring of FFEvaluate class

stefdoerr opened the pull request:
to be able to get all results and renamed methods.

I really dislike the name `run` in this case and also run only returned energies while ffevaluate calculates also forces and atom energies. So now `calculate` returns everything and `getEnergies` only the formatted energies.

j3mdamas commented on htmd/ffevaluation/ffevaluate.py:
```diff
@@ -26,19 +27,26 @@ def __init__(self, mol, prm, betweensets=None, cutoff=0, rfa=False, solventDiele
         args.append(solventDielectric)
         self._args = args
 
+    @_Deprecated('1.11.11', 'getEnergies')
```
omg, you're using my decorator!

stefdoerr replied:
:heart:

stefdoerr commented:
Opinions? I am thinking actualy of also deprecating the `ffevaluate` function. The initialization of the parameters takes the longest time so using it as a class which does the initialization in the `init` and then can calculate fast the energies for different coordinates forces people to use it the fast way. If you call 1000 times `ffevaluate` function it rebuilds the parameters 1000 times, while using the class you can call the `init` once and then only evaluate energies with the same parameters.

giadefa commented:
yes, it's better with a class.

On 20 March 2018 at 11:42, Stefan Doerr <notifications@github.com> wrote:

> Opinions? I am thinking actualy of also deprecating the ffevaluate
> function. The initialization of the parameters takes the longest time so
> using it as a class which does the initialization in the init and then
> can calculate fast the energies for different coordinates forces people to
> use it the fast way. If you call 1000 times ffevaluate function it
> rebuilds the parameters 1000 times, while using the class you can call the
> init once and then only evaluate energies with the same parameters.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Acellera/htmd/pull/600#issuecomment-374551560>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHaqOhSQ5Xcon6bm43gYXW0LwnsDcLkaks5tgN0igaJpZM4SwhD7>
> .
>



-- 
<http://www.acellera.com>

       <https://twitter.com/acellera>
<https://www.youtube.com/user/acelleracom>
<https://www.linkedin.com/company/2133167?trk=tyah&trkInfo=clickedVertical%3Acompany%2CclickedEntityId%3A2133167%2Cidx%3A2-1-2%2CtarId%3A1448018583204%2Ctas%3Aacellera>
    <https://www.acellera.com/md-simulation-blog-news/>
<http://is.gd/1eXkbS>

j3mdamas commented:
Well, if only there was an `FFMolecule` class ;)

Let's discuss the redesign of the parameterization stuff inside HTMD with @raimis?

stefdoerr commented:
How is this related to FFMolecule or parametrization?

On Tue, Mar 20, 2018 at 11:46 AM, João M. Damas <notifications@github.com>
wrote:

> Well, if only there was an FFMolecule class ;)
>
> Let's discuss the redesign of the parameterization stuff inside HTMD with
> @raimis <https://github.com/raimis>?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Acellera/htmd/pull/600#issuecomment-374552747>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHkVgrA7yY1X1_CcKMNkWwAQZTYYzf3kks5tgN4dgaJpZM4SwhD7>
> .
>

j3mdamas commented:
well, ffevalute is part of the parameterization tooling (even though not restricted to it). we decided that after the strip down refactoring we would try to design some structure for it, no?

stefdoerr commented:
I disagree. Parametrization *uses* (marginally even) FFevaluate but it is a
totally independent project.

On Tue, Mar 20, 2018 at 11:50 AM, João M. Damas <notifications@github.com>
wrote:

> well, ffevalute is part of the parameterization tooling (even though not
> restricted to it). we decided that after the strip down refactoring we
> would try to design some structure for it, no?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Acellera/htmd/pull/600#issuecomment-374553663>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHkVgo7v00Fkm_N3mif8jMIQRwr7e94pks5tgN7lgaJpZM4SwhD7>
> .
>

j3mdamas commented:
ok, considering the fitting is actually not done using FFevaluate, you may be right. but making it's place inside htmd should take a broader view, don't you agree?

stefdoerr commented:
I cleaned it up now, removing the function, adding docs and adding it to `htmd.ui`. I also renamed `getEnergies` to `calculateEnergies` to show that it doesn't just get but does actual calculations. Added also docs which clearly state that.

stefdoerr commented:
For me in the current state it's ok. Tell me opinions and if it passes we merge
